### ISSUE TITLE
Intermittent FgaApiAuthenticationError when using Oauth2 Client Credentials

### ISSUE THREAD

**Issue by nverbos-godaddy** [openfga/java-sdk#75]
### Checklist

- [X] I have looked into the [README](https://github.com/openfga/java-sdk/blob/main/README.md) and have not found a suitable solution or answer.
- [X] I have looked into the [documentation](https://openfga.dev/docs) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/openfga/java-sdk/issues) and have not found a suitable solution or answer.
- [X] I have upgraded to the [latest version](https://github.com/openfga/java-sdk/releases/latest) of OpenFGA and the issue still persists.
- [X] I have searched the [Slack community](https://openfga.dev/community) and have not found a suitable solution or answer.
- [X] I agree to the terms within the [OpenFGA Code of Conduct](https://github.com/openfga/.github/blob/main/CODE_OF_CONDUCT.md).

### Description

I have observed FgaApiAuthenticationError is thrown intermittently when using OAuth2 client credentials.  This occurs most frequently after a period of inactivity where the OpenFgaClient is not used for a long period of time, or at an intervals roughly equal to the OAuth2 access token's `expires_in` value.  I think there may be an issue with the way that OAuth2 tokens are refreshed before expiry.

### Expectation

I expect for OAuth2 access tokens to be refreshed automatically before being sent to the openfga api.  An expired token should never be sent to the openfga api and the openFgaClient does not intermittently throw `FgaApiAuthenticationError`

### Reproduction

**Given:**

Given I set up an OpenFgaClient using [the Oauth2 Credentials example in the README ](https://github.com/openfga/java-sdk?tab=readme-ov-file#oauth2-credentials)
```
        var config = new ClientConfiguration()
                .apiUrl(System.getenv("FGA_API_URL")) // If not specified, will default to "http://localhost:8080"
                .storeId(System.getenv("FGA_STORE_ID")) // Not required when calling createStore() or listStores()
                .authorizationModelId(System.getenv("FGA_AUTHORIZATION_MODEL_ID")) // Optional, can be overridden per request
                .credentials(new Credentials(
                    new ClientCredentials()
                            .apiTokenIssuer(System.getenv("FGA_API_TOKEN_ISSUER"))
                            .scopes(System.getenv("FGA_API_SCOPES")) // optional space separated scopes
                            .clientId(System.getenv("FGA_CLIENT_ID"))
                            .clientSecret(System.getenv("FGA_CLIENT_SECRET"))
                ));

        var openFgaClient = new OpenFgaClient(config);
```
        
        
**When:**
When I make an openfga check request using the openfgaClient set up using clientCredentials
```
openFgaClient.check(checkRequest)
```

**Then:**
- The openFgaClient always sends a fresh OAuth2 access token to the openfga server.
- The openFgaClient ensures that the OAuth2 access token is refreshed before being sent if it is expired
- The openFgaClient does not intermittently throw `FgaApiAuthenticationError`

### OpenFGA SDK version

0.4.1

### OpenFGA version

1.3.3

### SDK Configuration

OAuth2 Client Credentials

### Logs

```
dev.openfga.sdk.errors.FgaApiAuthenticationError: check
  at dev.openfga.sdk.errors.FgaError.getError(FgaError.java:49)
  at dev.openfga.sdk.api.client.HttpRequestAttempt.lambda$attemptHttpRequest$2(HttpRequestAttempt.java:56)
  at java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1150)
  at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
  at java.util.concurrent.CompletableFuture.postFire(CompletableFuture.java:614)
  at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:844)
  at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:482)
  at java.lang.Thread.run(Thread.java:833)
```

**Comment by rhamzeh** [comment]
Thanks for raising this @nverbos-godaddy!

**Comment by ewanharris** [comment]
Apologies for not following up here @nverbos-godaddy, we released a fix in v0.4.2 of the SDK last week. Please let us know if you see any further anomalies with the token refreshes
